src/main/java/org/reactome/qa/doisuggester/Main.java

```diff
@@ -50,213 +56,201 @@ public static void main(String[] args) throws FileNotFoundException, IOException
-						{
-							if (rleIE.getDBID().equals(pathwayIE.getDBID()))
-							{
-								// The parent Pathway contained the IE.
-								hasRLEInstanceEdit = true;
-							}
-						}
-						if (hasRLEInstanceEdit)
+
+						// At time of writing, no 'existing' RlEs have a new IE in the immediate parent
+						// when using
+						// 'test_slice_20210118' as the current DB, and 'slice_previous_as_74' as the
+						// previous DB.
+						if (hasRLEInstanceEdit(newIE, parentPathwayIE))
 						{
-							// If it has the new IE, it might be not be the highest Pathway in the hierarchy
-							// with it.
-							// Continue looking up the hierarchy to try and find the HIGHEST Pathway with
-							// the IE.
-							GKInstance grandparentPathwayWithRLEInstanceEdit = getGrandparentPathwayWithRLEInstanceEdit(parentPathway, rleIE);
-							Map<GKInstance, GKInstance> reaction2newIE = new HashMap<>();
-							reaction2newIE.put(newRLE, rleIE);
-							// If 'Pathway' key exists, add to it
-							if (pathway2Reactions2NewIEs.get(grandparentPathwayWithRLEInstanceEdit) != null)
-							{
-								pathway2Reactions2NewIEs.get(grandparentPathwayWithRLEInstanceEdit).add(reaction2newIE);
-							}
-							else
-							{
-								// If 'Pathway' key doesn't exist, create it and then add to it.
-								Set<Map<GKInstance, GKInstance>> reaction2newIESet = new HashSet<>(Arrays.asList(reaction2newIE));
-								pathway2Reactions2NewIEs.put(grandparentPathwayWithRLEInstanceEdit, reaction2newIESet);
-							}
+							System.out.println("This IE exists in immediate parent Pathway");
 						}
 						else
 						{
-							// If no IE exists in the immediate Parent, then we don't check up the hierarchy
-							// (correct?), but automatically assume that is the
-							// highest Pathway that needs to be updated.
-							Map<GKInstance, GKInstance> reaction2newIE = new HashMap<>();
-							reaction2newIE.put(newRLE, rleIE);
-							// If 'Pathway' key exists, add to it
-							if (pathway2Reactions2NewIEs.get(parentPathway) != null)
+
+							// Since the new IE wasn't found in the immediate parent, check up the Pathway
+							// hierarchy.
+							GKInstance parentPathwayWithRLEInstanceEdit = getGrandparentPathwayWithRLEInstanceEdit(parentPathway, newIE);
+
+							// If a Pathway with the IE was found, it is added to a map organized as
+							// {pathway => {reaction:newIE}}
+							if (parentPathwayWithRLEInstanceEdit != null)
 							{
-								pathway2Reactions2NewIEs.get(parentPathway).add(reaction2newIE);
+								addPathwayToMap(pathwaysMap, currentRLE, newIE, parentPathwayWithRLEInstanceEdit);
 							}
 							else
 							{
-								// If 'Pathway' key doesn't exist, create it and then add to it.
-								Set<Map<GKInstance, GKInstance>> reaction2newIESet = new HashSet<>(Arrays.asList(reaction2newIE));
-								pathway2Reactions2NewIEs.put(parentPathway, reaction2newIESet);
+								addPathwayToMap(pathwaysMap, currentRLE, newIE, parentPathway);
```

Comment:
This occurs is executed if the `parentPathway` is already known not to have the instance edit.  I'm confused why the `else` block here adds it to the pathway map?

Reply:
This is a matter of the program's logic. I don't fully understand what Justin did either, and I am considering a major structural reorganisation.